plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/CabalAdd.hs

```diff
@@ -278,6 +278,9 @@ getDependencyEdit recorder env cabalFilePath buildTarget dependency = do
       pure edit
 
 -- | Given a path to a haskell file, returns the closest cabal file.
+--   If package.yaml is present in same directory as .cabal file, it returns nothing, because adding a dependency to a generated cabal file
+--   will break propagation of changes from package.yaml to cabal file in stack project.
+--
```

Comment:
```suggestion
--   If a package.yaml is present in same directory as the .cabal file, returns nothing, because adding a dependency to a generated cabal file
--   will break propagation of changes from package.yaml to cabal files in stack projects.
```